File: poh4/poh4_3.ex
#!/usr/bin/env elixir
a=IO.gets("") |> String.trim() |> String.split(" ")
t=Enum.at(a,0) |> String.to_integer()
n=Enum.at(a,1) |> String.to_integer()
a=List.to_tuple(Enum.map(0..n-1, fn _ ->
	IO.gets("")|>String.trim()|>String.to_integer()
end))
cur=0..t-1 |> Enum.reduce(0, fn i, acc ->
	acc+elem(a,i)
end)
r={cur,cur}
if t<=n-1 do
	r=t..n-1 |> Enum.reduce(r, fn i, acc ->
		cur=elem(acc,1)+elem(a,i)-elem(a,i-t)
		{max(elem(acc,0),cur),cur}
	end)
	IO.puts(elem(r,0))
else
	IO.puts(cur)
end

File: poh4/poh4_3.rb
#!/usr/bin/ruby
T,N=gets.split.map(&:to_i)
a=Array.new(N)
a[0]=gets.to_i
cur=a[0]
i=1

if N==300000 && T==150000 # 4/5
elsif N==120 && T==10 #2
elsif N==1 && T==1 # 3
elsif N==12 && T==4 # 1
end

while i<T
	a[i]=gets.to_i
	cur+=a[i]
	i+=1
end
r=cur
while i<N
	a[i]=gets.to_i
	cur+=a[i]
	cur-=a[i-T]
	r=cur if r<cur
	i+=1
end
p r
